dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -860,141 +857,154 @@ protected SolrQuery resolveToSolrQuery(Context context, DiscoverQuery discoveryQ
+                        //TODO: do not assume this, people may want to use it for other ends, use a regex to make sure
+                        //We have a facet query, the values looks something like:
+                        //dateissued.year:[1990 TO 2000] AND -2000
+                        //Prepare the string from {facet.field.name}:[startyear TO endyear] to startyear - endyear
+                        String facetField = facetQuery.substring(0, facetQuery.indexOf(":"));
+                        String name = "";
+                        String filter = "";
+                        if (facetQuery.indexOf('[') > -1 && facetQuery.lastIndexOf(']') > -1) {
+                            name = facetQuery.substring(facetQuery.indexOf('[') + 1);
+                            name = name.substring(0, name.lastIndexOf(']')).replaceAll("TO", "-");
+                            filter = facetQuery.substring(facetQuery.indexOf('['));
+                            filter = filter.substring(0, filter.lastIndexOf(']') + 1);
+                        }
+                        Integer count = sortedFacetQueries.get(facetQuery);
+
+                        //No need to show empty years
+                        if (0 < count) {
+                            result.addFacetResult(facetField,
+                                                  new DiscoverResult.FacetResult(filter, name, null, name, count,
+                                                                                 DiscoveryConfigurationParameters
+                                                                                     .TYPE_DATE));
+                        }
                     }
+                }
 
-                    Integer count = sortedFacetQueries.get(facetQuery);
-
-                    //No need to show empty years
-                    if (0 < count) {
-                        result.addFacetResult(facetField,
-                                              new DiscoverResult.FacetResult(filter, name, null, name, count,
-                                                                             DiscoveryConfigurationParameters
-                                                                                 .TYPE_DATE));
+                if (solrQueryResponse.getSpellCheckResponse() != null) {
+                    String recommendedQuery = solrQueryResponse.getSpellCheckResponse().getCollatedResult();
+                    if (StringUtils.isNotBlank(recommendedQuery)) {
+                        result.setSpellCheckQuery(recommendedQuery);
                     }
                 }
             }
-
-            if (solrQueryResponse.getSpellCheckResponse() != null) {
-                String recommendedQuery = solrQueryResponse.getSpellCheckResponse().getCollatedResult();
-                if (StringUtils.isNotBlank(recommendedQuery)) {
-                    result.setSpellCheckQuery(recommendedQuery);
-                }
+            if (zombieDocs.size() != 0) {
```

Comment:
We should comment what is going on here. Something like: "If any stale entries are found in the current page of results, we remove those stale entries and rerun the same query again.  Otherwise, the query is valid and the results are returned."